Fix Java manual instrumentation with Spring Boot app

PerfectSlayer opened the pull request:
Hi OpenTelemetry team 👋 

Following [a dependency management issue using Java Spring Boot app and Gradle build system](https://github.com/open-telemetry/opentelemetry.io/issues/3613), I updated the _Java Manual Instrumentation_ documentation page.

This PR proposes a fix to https://github.com/open-telemetry/opentelemetry.io/issues/3613 and the rationale for its content (using OTel BOM import rather than removing Spring dependency management pluging) is explained in my comment: https://github.com/open-telemetry/opentelemetry-java/issues/6018#issuecomment-1842561406.

It's my first contribution to this repository so feel free to provide guidance if I miss something. 
I have built the site locally and tested it using `npm run test`. It seemed fine 😇 

>[!NOTE]
> In addition, I updated the highlight sections, fix semconv versions and restore a missing semi-column.

jeanbisutti commented:
@PerfectSlayer There are [two ways](https://docs.spring.io/spring-boot/docs/current/gradle-plugin/reference/htmlsingle/#managing-dependencies) to manage a BOM with Gradle and Spring Boot. Could you please also document the way with the  Gradle’s native bom support (as I did on [this PR](https://github.com/open-telemetry/opentelemetry.io/pull/3672) for the Spring Boot page)?

jeanbisutti commented:
@PerfectSlayer Could you also please add documentation on how to configure the things with Maven (as I did on https://github.com/open-telemetry/opentelemetry.io/pull/3672 for the Spring Boot page)?

svrnm commented on content/en/docs/instrumentation/java/manual.md:
````diff
@@ -192,10 +192,16 @@ dependencies for the OpenTelemetry API.
 
-```kotlin { hl_lines=3 }
+```kotlin { hl_lines=["1-5",9] }
+dependencyManagement {
+	imports {
+		mavenBom("io.opentelemetry:opentelemetry-bom:{{% param vers.otel %}}")
+	}
+}
````
```suggestion
dependencyManagement {
    imports {
        mavenBom("io.opentelemetry:opentelemetry-bom:{{% param vers.otel %}}")
    }
}
```

svrnm commented:
@open-telemetry/java-approvers PTAL

PerfectSlayer commented:
@jeanbisutti Sure, I was waiting for #3672 to be reviewed first.

PerfectSlayer commented:
It pushed some changes to reflect BOM documentation from @jeanbisutti 
I also moved a paragraph (_throughout this documentation you will add dependencies.…_) from the Gradle tab below, to be accessible for Maven users too, and removed duplicated maven dependencies to stay consistent (only adding SDK on installing SDK chapter, like autoconfigure).

jeanbisutti commented on content/en/docs/instrumentation/java/manual.md:
```diff
@@ -190,22 +190,39 @@ You should get a list of 12 numbers in your browser window, for example:
-artifact coordinates, see [releases]. For semantic convention releases, see
-[semantic-conventions-java].
+If you are not using Spring and its `io.spring.dependency-management` dependency
+managemnt plugin, install the OpenTelemetry BOM and API using Gradle depenencies
```
```suggestion
managemnt plugin, install the OpenTelemetry BOM and API using Gradle dependencies
```

cartermp commented on content/en/docs/instrumentation/java/manual.md:
```diff
@@ -190,22 +190,39 @@ You should get a list of 12 numbers in your browser window, for example:
-artifact coordinates, see [releases]. For semantic convention releases, see
-[semantic-conventions-java].
+If you are not using Spring and its `io.spring.dependency-management` dependency
+managemnt plugin, install the OpenTelemetry BOM and API using Gradle dependencies
```
```suggestion
management plugin, install the OpenTelemetry BOM and API using Gradle dependencies
```

cartermp commented:
@PerfectSlayer if you run `npm run fix:format` then this should be good.

PerfectSlayer commented:
@cartermp I run it and amend the commit it fixed.

cartermp commented:
@open-telemetry/java-approvers please take a look!

PerfectSlayer commented:
Hey there! Happy New Year 🎉 

Coming to check up on things. Is there anything I can do on my side to get this PR progress?

trask commented:
we are good to go from the Java SIG side with @laurit and @jeanbisutti's approvals

jack-berg commented on content/en/docs/instrumentation/java/manual.md:
````diff
@@ -190,22 +190,39 @@ You should get a list of 12 numbers in your browser window, for example:
 ```
 
-Throughout this documentation you will add dependencies. For a full list of
-artifact coordinates, see [releases]. For semantic convention releases, see
````
> Throughout this documentation you will add dependencies. For a full list of
artifact coordinates, see [releases].

I like this section near the top - maybe even the first line in the dependency management section. Starting with the bom feels strange without first establishing which artifacts the bom aligns.

jack-berg commented on content/en/docs/instrumentation/java/manual.md:
```diff
@@ -233,47 +250,40 @@ artifact coordinates, see [releases]. For semantic convention releases, see
-If you instrument a Java app, install the dependencies for the OpenTelemetry
-SDK.
+If you instrument a Java app, install the dependencies for the OpenTelemetry SDK
+in addition to the dependencies for the OpenTelemetry API.
```
The phrasing here is odd. Your change is fine but I wonder if we should correct it. 

> If you instrument a Java app, install the dependencies for the OpenTelemetry SDK

That's a really poor introduction to the SDK. Maybe something like:

"The OpenTelemetry API provides a set of interfaces for collecting telemetry, but the data is dropped without an implementation. The OpenTelemetry SDK is the implementation of the OpenTelemetry API provided by OpenTelemetry. To use it, begin by installing dependencies:"

jack-berg commented on content/en/docs/instrumentation/java/manual.md:
```diff
@@ -233,47 +250,40 @@ artifact coordinates, see [releases]. For semantic convention releases, see
+    implementation("io.opentelemetry:opentelemetry-sdk");
+    implementation("io.opentelemetry:opentelemetry-sdk-metrics");
+    implementation("io.opentelemetry:opentelemetry-exporter-logging");
+    implementation("io.opentelemetry:opentelemetry-semconv:{{% param vers.semconv %}}-alpha");
```
```suggestion
    implementation("io.opentelemetry.semconv:opentelemetry-semconv:{{% param vers.semconv %}}-alpha");
```
As documented here: https://github.com/open-telemetry/semantic-conventions-java

jack-berg commented on content/en/docs/instrumentation/java/manual.md:
```diff
@@ -233,47 +250,40 @@ artifact coordinates, see [releases]. For semantic convention releases, see
-    implementation("io.opentelemetry:opentelemetry-semconv:{{% param vers.otel %}}-alpha");
+    implementation("io.opentelemetry:opentelemetry-api");
+    implementation("io.opentelemetry:opentelemetry-sdk");
+    implementation("io.opentelemetry:opentelemetry-sdk-metrics");
```
```suggestion
```
This dependency is redundant.

jack-berg commented on content/en/docs/instrumentation/java/manual.md:
```diff
@@ -233,47 +250,40 @@ artifact coordinates, see [releases]. For semantic convention releases, see
-        </dependency>
         <dependency>
             <groupId>io.opentelemetry</groupId>
             <artifactId>opentelemetry-sdk</artifactId>
```
The `opentelemetry-sdk-metrics` dependency below is redundant.

jack-berg commented on content/en/docs/instrumentation/java/manual.md:
```diff
@@ -233,47 +250,40 @@ artifact coordinates, see [releases]. For semantic convention releases, see
-    implementation("io.opentelemetry:opentelemetry-sdk-metrics:{{% param vers.otel %}}");
-    implementation("io.opentelemetry:opentelemetry-exporter-logging:{{% param vers.otel %}}");
-    implementation("io.opentelemetry:opentelemetry-semconv:{{% param vers.otel %}}-alpha");
+    implementation("io.opentelemetry:opentelemetry-api");
```
```suggestion
```
I believe your intent is to omit `opentelemetry-api` here since its included above.

jack-berg commented on content/en/docs/instrumentation/java/manual.md:
```diff
@@ -312,16 +322,16 @@ To use auto-configuration add the following dependency to your application:
-    implementation("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi:{{% param vers.otel %}}");
+    implementation("io.opentelemetry:opentelemetry-api");
+    implementation("io.opentelemetry:opentelemetry-sdk");
+    implementation("io.opentelemetry:opentelemetry-sdk-metrics");
```
```suggestion
```
Redundant

jack-berg commented on content/en/docs/instrumentation/java/manual.md:
```diff
@@ -312,16 +322,16 @@ To use auto-configuration add the following dependency to your application:
+    implementation("io.opentelemetry:opentelemetry-exporter-logging");
+    implementation("io.opentelemetry.semconv:opentelemetry-semconv:{{% param vers.semconv %}}-alpha");
+    implementation("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure");
+    implementation("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi");
```
```suggestion
    implementation("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi");
```
Redundant

jack-berg reviewed (commented):
Left a handful of comments that aren't necessarily related to the changes @PerfectSlayer makes, but didn't seem right to let them continue to go unnoticed.

svrnm commented:
@PerfectSlayer can you take a look at the review comments?

PerfectSlayer commented:
Sure will do this tomorrow or next week 👍

zeitlinger commented:
> In addition, I updated the highlight sections

what is the intention of the highlights?

chalin commented on content/en/docs/instrumentation/java/manual.md:
````diff
@@ -190,22 +190,39 @@ You should get a list of 12 numbers in your browser window, for example:
 
-```kotlin { hl_lines=3 }
+```kotlin { hl_lines=["1-5",9] }
+	  imports {
+		    mavenBom("io.dependencyManagement {
+opentelemetry:opentelemetry-bom:{{% param vers.otel %}}")
+	  }
````
Indentation / formatting looks off here.

PerfectSlayer replied:
Good catch!

chalin reviewed (commented):
See inline comment.

This will need to be rebased.

PerfectSlayer commented:
I rebased the branch and should have taken all the review comments into account.

jeanbisutti commented:
LGTM

Thanks a lot @PerfectSlayer!

It would remain the comment from Gregor: https://github.com/open-telemetry/opentelemetry.io/pull/3656#issuecomment-1908086538 @zeitlinger, could you please elaborate a bit your question?

svrnm commented:
@PerfectSlayer can you resolve the conflict?

PerfectSlayer commented:
@svrnm I just did using GH Web UI.
